docs/docs-content/user-management/authentication/api-key/delete-api-key.md

```diff
@@ -92,3 +96,248 @@ learn more about deleting an API key as a tenant admin.
+           "deletionTimestamp": "0001-01-01T00:00:00.000Z",
+           "lastModifiedTimestamp": "2024-09-16T14:46:29.079Z",
+           "name": "remove-me-test",
+           "uid": "66e844c44bab2337f20c7471"
+         },
+         "spec": {
+           "expiry": "2024-09-23T14:46:28.164Z",
+           "user": {
+             "firstName": "example",
+             "lastName": "example",
+             "uid": "*****************"
+           }
+         },
+         "status": {
+           "isActive": true
+         }
+       }
+     ]
+   }
+   ```
+
+3. Once you have the API key's unique identifier, issue the following command to delete the API key. Replace `uid` with
+   the API key's unique identifier. Specify a valid API key in the `ApiKey` header.
+
+   ```shell
+   curl -L -X DELETE 'https://api.spectrocloud.com/v1/apiKeys/:uid' \
+   -H 'ApiKey: <API_KEY_VALUE>'
+   ```
+
+4. No output is expected if the API key is successfully deleted.
+
+### Validate
+
+1. Verify the API key is no longer available in the Palette by issuing the following command. Replace `API_KEY_VALUE`
+   with your API key.
+
+   ```shell
+   curl --location 'https://api.spectrocloud.com/v1/apiKeys' \
+   --header 'Accept: application/json' \
+   --header 'apiKey: API_KEY_VALUE'
+   ```
+
+2. The API key should not be listed in the response. If the API key is still available, verify the API key's unique
+   identifier and reissue the delete command. You can also validate the deletion by checking the Palette UI.
+
+## SDK
+
+You can use the [Palette SDK](../../../automation/palette-sdk/palette-sdk.md) to delete an API key programmatically.
+
+### Prerequisites
+
+- You must have a valid Palette API key. Refer to the [Create API Key](create-api-key.md) section for more information.
+
+- [Go version](https://go.dev/doc/install) 1.22 or later.
+
+- A text editor or an IDE to write and execute the Go code.
+
+- A valid Palette API key to delete. In this example, the fictional API key named `delete-test-key` is used.
+
+- An internet connection to download the Palette SDK and its dependencies.
+
+### Delete API Key With Go SDK
+
+1. Create a new directory for your Go project and navigate to the directory.
+
+   ```shell
+   mkdir delete-api-key && cd delete-api-key
+   ```
+
+2. Create a new Go file, for example, **main.go**.
+
+   ```shell
+   touch main.go
+   ```
+
+3. Initialize the Go module. Use the following command to initialize the Go module.
+
+   ```shell
+   go mod init example/delete-api-key
+   ```
+
+4. Open the **main.go** file in your text editor or IDE.
+
+5. Copy and paste the following code snippet into the **main.go** file.
+
+   ```go
+    package main
+
+    import (
+      "fmt"
+      "log"
+      "log/slog"
+      "os"
+
+      "github.com/spectrocloud/palette-sdk-go/client"
+    )
+
+    func main() {
+
+      host := os.Getenv("PALETTE_HOST") // "api.spectrocloud.com"
+      apiKey := os.Getenv("PALETTE_API_KEY") // "your api key"
```

Comment:
Should we exit the program if host and apiKey are not set?

Reply:
Yeah, we should.